grid-client/deployer/network_deployer.go

```diff
@@ -274,7 +275,7 @@ func (d *NetworkDeployer) Deploy(ctx context.Context, znet *workloads.ZNet) erro
 		return err
 	}
 
-	newDeployments, err := d.GenerateVersionlessDeployments(ctx, znet)
+	newDeployments, err := d.GenerateVersionlessDeployments(ctx, znet, nil)
```

Comment:
What if I deployed some networks sequentially, how can I track the used ports?

Reply:
we could add it to both Deploy and BatchDeploy but wouldn't the user just use BatchDeploy if they are deploying multiple networks?

Reply:
Would it be better if we saved it in the network state?

Reply:
I don't think we need store this data between deployments as the ports data are fetched from the node itself. so after deploying a network, the second time we deploy a network the ports of the first network will be fetched form the node itself (among other ports). so I don't see it needed outside of deploying a batch of networks.